service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -60,18 +67,10 @@ public DigitalTwinAsyncClient(@NonNull String connectionString) {
      * @param <T> The generic type to deserialize the application/json into.
      * @return The application/json of the digital twin.
      */
-    public <T> Observable<T> getDigitalTwin (@NonNull String digitalTwinId, Class<T> clazz)
+    public <T> Observable<T> getDigitalTwin (String digitalTwinId, Class<T> clazz)
     {
-        return digitalTwin.getDigitalTwinAsync(digitalTwinId)
-                .filter(Objects::nonNull)
-                .flatMap(response -> {
-                    try {
-                        return Observable.just(DeserializationHelpers.castObject(objectMapper, response, clazz));
-                    } catch (JsonProcessingException e) {
-                        return Observable.error(e);
-                    }
-                })
-                .subscribeOn(Schedulers.io());
+        return getDigitalTwinWithResponse(digitalTwinId, clazz)
+                .map(response -> response.body());
```

Comment:
nit: lambda can be replaced by method reference

Reply:
This seemed more readable to me, specially because it is a small thing to do inside of map.